pyqtgraph/graphicsItems/PlotItem/PlotItem.py

```diff
@@ -1169,6 +1172,58 @@ def showAxis(self, axis, show=True):
     def hideAxis(self, axis):
         """Hide one of the PlotItem's axes. ('left', 'bottom', 'right', or 'top')"""
         self.showAxis(axis, False)
+        
+    def showAxes(self, selection, showValues=True, size=None):
+        """ 
+        Convenience method for quickly configuring axis settings.
+        
+        Parameters
+        ----------
+        selection: boolean or tuple of booleans (left, top, right, bottom)
+            Determines which AxisItems will be displayed.
+            A single boolean value will set all axes, 
+            so that ``showAxes(True)`` configures the axes to draw a frame.
+        showValues: optional, boolean or tuple of booleans (left, top, right, bottom)
+            Determines if values will be displayed for the ticks of each axis.
+            True value shows values for left and bottom axis (default).
+            False shows no values.
+            None leaves settings unchanged.
+            If not specified, left and bottom axes will be drawn with values.
+        size: optional, float or tuple of floats (width, height)
+            Reserves as fixed amount of space (width for vertical axis, height for horizontal axis)
+            for each axis where tick values are enabled. If only a single float value is given, it
+            will be applied for both width and height.
+        """
+        if selection is True:
+            selection = (True, True, True, True)
+        elif selection is False:
+            selection = (False, False, False, False)
+        if showValues is True:
+            showValues = (True, False, False, True)
+        elif showValues is False:
+            showValues = (False, False, False, False)
+        elif showValues is None:
+            showValues = (None, None, None, None)
+        if size is None:
+            size = (None, None)
+        elif not isinstance(size, collections.abc.Sized):
+            size = (size, size)
+        for idx, key in enumerate(('left','top','right','bottom')):
+            if selection[idx] is True : self.showAxis(key)
+            if selection[idx] is False: self.hideAxis(key)
```

Comment:
can we redo these lines by having

```python
self.showAxis(key) if selection[idx] else self.hideAxis(key)
```

Reply:
I think we can't because this is really a three-way choice:
`True` means turn on
`False` means turn off
`None` (or anything else, really) means leave as is.

Reply:
ahh, you're right, that cannot be replicated with the ternary operator.

Probably should do `if/elif/else` then

```python
for idx, key in enumerate(('left', 'top', 'right', 'bottom')):
    if selection[idx] is True:
        self.showAxis(key)
    elif selection[idx] is False:
        self.hideAxis(key)
    else:
        continue # don't do anything otherwise
    ...
```

Reply:
actually, we can redo this entire loop now that I look at it...

```python
for idx, key in enumerate(('left','top','right','bottom')):
    ax = self.getAxis(key)
    if selection[idx] is True: 
        self.showAxis(key)
        ax.setStyle(showValues=True )
            if key in ('left','right') and size[0] is not None:
                ax.setWidth( size[0] )
            elif key in ('top', 'bottom') and size[1] is not None:
                ax.setHeight( size[1] )
    elif selection[idx] is False:
        self.hideAxis(key)
        ax.setStyle(showValues=False )
        if key in ('left','right') and size[0] is not None:
            ax.setWidth( None ) # revert to auto-width
        elif key in ('top', 'bottom') and size[1] is not None:
            ax.setHeight( None ) # revert to auto-height
    else:
        continue  # if selection[idx] is not True or False do nothing...
```

This gets us out of doing the same if-check `showValues[idx] is ...` multiple times.

Reply:
I may be missing something here, but some of the repeated code handles the separation of displaying just the axis and the text labels associated with the axis.
We could maybe shorten the code a little bit by only handling the labels when the axis is actually visible. But since the axis is never really deleted, it still has a well-defined state even if it is not shown. So I think there is less potential for unexpected behavior if we always configure the labels as requests, visible or not.